Makefile

```diff
@@ -121,8 +121,16 @@ PRIV_TEST_PKGS ?= $(PRIV_TEST_PKGS_EVAL)
 tests-privileged:
 	# cilium-map-migrate is a dependency of some unit tests.
 	$(QUIET) $(MAKE) $(SUBMAKEOPTS) -C bpf cilium-map-migrate
-	$(QUIET)$(foreach pkg,$(PRIV_TEST_PKGS),\
-		$(GO_TEST) $(TEST_LDFLAGS) github.com/cilium/cilium/$(pkg) $(GOTEST_PRIV_OPTS) || exit 1;)
+
+	$(QUIET) echo "mode: count" > coverage-all-tmp.out
+	$(QUIET) echo "mode: count" > coverage.out
+	for pkg in $(patsubst %,github.com/cilium/cilium/%,$(PRIV_TEST_PKGS)); do \
+		$(GO_TEST) $(TEST_LDFLAGS) $$pkg $(GOTEST_PRIV_OPTS) $(GOTEST_COVER_OPTS) \
+		|| exit 1; \
+		tail -n +2 coverage.out >> coverage-all-tmp.out; \
+	done
+	$(MAKE) generate-cov
+	@rmdir ./daemon/1 ./daemon/1_backup 2> /dev/null || true
```

Comment:
Did you find that this step was necessary here as well or is it just following the existing code?

Also wondering (1) why this even exists; and if it needs to, (2) why not put it as part of the `generate-cov` target.

Reply:
I was following existing code. I was trying to do some git archaeology to understand why it was there to begin with, but I was only able to get to this commit (33244acae) which doesn't explain why it's needed. I presume some tests left state and needed to be cleaned up, and I didn't go further than that.

Anyway, moving it to the `generate-cov` target is a good idea.